index.js

```diff
@@ -105,63 +105,22 @@ class DeltaChat extends EventEmitter {
 
     this.setConfig('addr', opts.addr)
 
-    if (typeof opts.mail_server === 'string') {
-      this.setConfig('mail_server', opts.mail_server)
-    }
-
-    if (typeof opts.mail_user === 'string') {
-      this.setConfig('mail_user', opts.mail_user)
-    }
-
-    if (opts.mail_port) {
-      this.setConfig('mail_port', String(opts.mail_port))
-    }
+    this.setConfig('mail_server', opts.mail_server)
+    this.setConfig('mail_user', opts.mail_user)
+    this.setConfig('mail_port', String(opts.mail_port))
 
     this.setConfig('mail_pw', opts.mail_pw)
-
-    if (typeof opts.send_server === 'string') {
-      this.setConfig('send_server', opts.send_server)
-    }
-
-    if (typeof opts.send_user === 'string') {
-      this.setConfig('send_user', opts.send_user)
-    }
-
-    if (typeof opts.send_pw === 'string') {
-      this.setConfig('send_pw', opts.send_pw)
-    }
-
-    if (opts.send_port) {
-      this.setConfig('send_port', String(opts.send_port))
-    }
-
-    if (opts.server_flags) {
-      this.setConfig('server_flags', String(opts.server_flags))
-    }
-
-    if (typeof opts.displayname === 'string') {
-      this.setConfig('displayname', opts.displayname)
-    }
-
-    if (typeof opts.selfstatus === 'string') {
-      this.setConfig('selfstatus', opts.selfstatus)
-    }
-
-    if (typeof opts.selfavatar === 'string') {
-      this.setConfig('selfavatar', opts.selfavatar)
-    }
-
-    if (typeof opts.e2ee_enabled === 'boolean') {
-      this.setConfig('e2ee_enabled', String(opts.e2ee_enabled ? 1 : 0))
-    }
-
-    if (typeof opts.mdns_enabled === 'boolean') {
-      this.setConfig('mdns_enabled', String(opts.mdns_enabled ? 1 : 0))
-    }
-
-    if (opts.save_mime_headers === true) {
-      this.setConfig('save_mime_headers', '1')
-    }
+    this.setConfig('send_server', opts.send_server)
+    this.setConfig('send_user', opts.send_user)
+    this.setConfig('send_pw', opts.send_pw)
+    this.setConfig('send_port', String(opts.send_port))
+    this.setConfig('server_flags', String(opts.server_flags))
+    this.setConfig('displayname', opts.displayname)
+    this.setConfig('selfstatus', opts.selfstatus)
+    this.setConfig('selfavatar', opts.selfavatar)
+    this.setConfig('e2ee_enabled', String(opts.e2ee_enabled ? 1 : 0))
```

Comment:
This means that `e2ee_enabled` will default to `false` which might be a problem, since in core it defaults to true.

Reply:
Also, the c side cries when passing `undefined`. We should probably add a guard in `setConfig()`, like:

```js
setConfig (key, value) {                                     
  return binding.dcn_set_config(this.dcn_context, key || '', value || '')
}                                                            
```

Reply:
@ralphtheninja but @r10s says that NULL should be accepted by core.

This is to implement the `advanced` settings in the UI. If you set some option, and then want to remove it later, you have no way of doing that now.

Reply:
> @ralphtheninja but @r10s says that NULL should be accepted by core.

Aye. So we already do this for other functions, but pass empty string and check for it on the c side, e.g. https://github.com/deltachat/deltachat-node/blob/master/src/module.c#L695